test/e2e/node_feature_discovery_test.go

```diff
@@ -1064,6 +1064,229 @@ resyncPeriod: "1s"
 					Expect(err).NotTo(HaveOccurred())
 				})
 			})
+
+			Context("selected namespaces restriction is respected or not", func() {
+				BeforeEach(func(ctx context.Context) {
+					extraMasterPodSpecOpts = []testpod.SpecOption{
+						testpod.SpecWithConfigMap("nfd-master-conf", "/etc/kubernetes/node-feature-discovery"),
+					}
+					cm := testutils.NewConfigMap("nfd-master-conf", "nfd-master.conf", `
+restrictions:
+  nodeFeatureNamespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: "fake"
+`)
+					_, err := f.ClientSet.CoreV1().ConfigMaps(f.Namespace.Name).Create(ctx, cm, metav1.CreateOptions{})
+					Expect(err).NotTo(HaveOccurred())
+				})
+				It("Nothing should be created", func(ctx context.Context) {
+					// deploy node feature object
+					if !useNodeFeatureApi {
+						Skip("NodeFeature API not enabled")
+					}
+
+					nodes, err := getNonControlPlaneNodes(ctx, f.ClientSet)
+					Expect(err).NotTo(HaveOccurred())
+
+					targetNodeName := nodes[0].Name
+					Expect(targetNodeName).ToNot(BeEmpty(), "No suitable worker node found")
+
+					// Apply Node Feature object
+					By("Creating NodeFeature object")
+					nodeFeatures, err := testutils.CreateOrUpdateNodeFeaturesFromFile(ctx, nfdClient, "nodefeature-1.yaml", f.Namespace.Name, targetNodeName)
+					Expect(err).NotTo(HaveOccurred())
+
+					By("Verifying node labels from NodeFeature object #1 are not created")
+					// No labels should be created since the f.Namespace is not in the selected Namespaces
+					expectedLabels := map[string]k8sLabels{
+						targetNodeName: {},
+						"*":            {},
+					}
+					eventuallyNonControlPlaneNodes(ctx, f.ClientSet).Should(MatchLabels(expectedLabels, nodes))
```

Comment:
Hmm, is this kinda racy as this would/could fall through even if the labels et al would eventually be created by NFD, but it hadn't yet created them at the time of this check?

Reply:
Should we add a sleep to wait, or what do you suggest to mitigate this racy case?

Reply:
Wouldn't [this](https://github.com/kubernetes-sigs/node-feature-discovery/pull/1592#discussion_r1668783915)? Check that thing X (e.g. annotations) was created and then check that thing Y (e.g. labels) wasn't